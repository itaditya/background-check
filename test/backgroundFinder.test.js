const backgroundFinder = require('../lib/backgroundFinder')

/* -- Mocks -- */
const extractRepoDetails = jest.fn().mockReturnValue({ owner: 'itaditya', repo: 'private-gh-app-test-repo' })
const getUserDiscussionIssue = jest.fn().mockReturnValue({
  data: {
    total_count: 0
  }
})
const getUserCommentedIssues = jest.fn().mockReturnValue({
  data: {
    total_count: 3,
    items: [{
      number: 15,
      repository_url: 'https://api.github.com/repos/aps120797/playground',
      url: 'https://api.github.com/repos/aps120797/playground/issues/17',
      html_url: 'https://github.com/aps120797/playground/issues/17'
    }, {
      number: 15,
      repository_url: 'https://api.github.com/repos/aps120797/playground',
      url: 'https://api.github.com/repos/aps120797/playground/issues/17',
      html_url: 'https://github.com/aps120797/playground/issues/17'
    }, {
      number: 15,
      repository_url: 'https://api.github.com/repos/aps120797/playground',
      url: 'https://api.github.com/repos/aps120797/playground/issues/17',
      html_url: 'https://github.com/aps120797/playground/issues/17'
    }]
  }
})
const getCommentsOnIssue = jest.fn().mockReturnValue({
  data: [{
    body: 'this project is the worst out there',
    user: {
      type: 'User',
      login: 'itaditya'
    }
  }]
})
const sentimentAnalyser = jest.fn().mockReturnValue(0.7)
const createDiscussionIssue = jest.fn().mockReturnValue(Promise.resolve({
  status: 200,
  data: {
    html_url: 'https://github.com/itaditya/maintainers-discussion/issues/17'
  }
}))
/* ^- Mocks -^ */

const backgroundFinderInstance = backgroundFinder({
  dependencies: {
    extractRepoDetails,
    getUserDiscussionIssue,
    getUserCommentedIssues,
    getCommentsOnIssue,
    sentimentAnalyser,
    createDiscussionIssue
  }
})

test('backgroundFinder is working', async () => {
  const context = {
    payload: {
      repository: { owner: { login: 'itaditya' } },
      comment: { user: { login: 'itaditya' } }
    },
    log: console.log
  }
  await backgroundFinderInstance(context)
  expect(extractRepoDetails.mock.calls).toMatchSnapshot()
  expect(getUserDiscussionIssue.mock.calls).toMatchSnapshot()
  expect(getUserCommentedIssues.mock.calls).toMatchSnapshot()
  expect(getCommentsOnIssue.mock.calls).toMatchSnapshot()
  expect(sentimentAnalyser.mock.calls).toMatchSnapshot()
  expect(createDiscussionIssue.mock.calls).toMatchSnapshot()
})
