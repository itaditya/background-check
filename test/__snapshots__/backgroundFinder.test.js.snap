// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backgroundFinder is working 1`] = `
Array [
  Array [
    "https://api.github.com/repos/aps120797/playground",
  ],
  Array [
    "https://api.github.com/repos/aps120797/playground",
  ],
  Array [
    "https://api.github.com/repos/aps120797/playground",
  ],
]
`;

exports[`backgroundFinder is working 2`] = `
Array [
  Array [
    undefined,
    Object {
      "owner": "itaditya",
      "username": "itaditya",
    },
  ],
]
`;

exports[`backgroundFinder is working 3`] = `
Array [
  Array [
    undefined,
    Object {
      "username": "itaditya",
    },
  ],
]
`;

exports[`backgroundFinder is working 4`] = `
Array [
  Array [
    undefined,
    Object {
      "issueNum": 15,
      "owner": "itaditya",
      "repo": "private-gh-app-test-repo",
    },
  ],
  Array [
    undefined,
    Object {
      "issueNum": 15,
      "owner": "itaditya",
      "repo": "private-gh-app-test-repo",
    },
  ],
  Array [
    undefined,
    Object {
      "issueNum": 15,
      "owner": "itaditya",
      "repo": "private-gh-app-test-repo",
    },
  ],
]
`;

exports[`backgroundFinder is working 5`] = `
Array [
  Array [
    "this project is the worst out there",
  ],
  Array [
    "this project is the worst out there",
  ],
  Array [
    "this project is the worst out there",
  ],
]
`;

exports[`backgroundFinder is working 6`] = `
Array [
  Array [
    undefined,
    Object {
      "owner": "itaditya",
      "toxicComments": Array [
        Object {
          "link": "https://github.com/aps120797/playground/issues/17",
          "text": "this project is the worst out there",
          "toxicScore": 0.7,
        },
        Object {
          "link": "https://github.com/aps120797/playground/issues/17",
          "text": "this project is the worst out there",
          "toxicScore": 0.7,
        },
        Object {
          "link": "https://github.com/aps120797/playground/issues/17",
          "text": "this project is the worst out there",
          "toxicScore": 0.7,
        },
      ],
      "username": "itaditya",
    },
  ],
]
`;
